examples/tutorial_2_customproto.md

```diff
@@ -0,0 +1,82 @@
+In the [previous tutorial](tutorial_1_connect.md), we've looked at how to create a simple ping program using the `nim-libp2p`.
+
+We'll now look at how to create a custom protocol inside the libp2p
+
+# Custom protocol in libp2p
+Let's create a `part2.nim`, and import our dependencies:
+```nim
+import bearssl
+import chronos
+import stew/byteutils
+
+import libp2p
+```
+This is similar to the first tutorial, except we don't need to import the `Ping` protocol.
+
+Next, we'll declare our custom protocol
+```nim
+const TestCodec = "/test/proto/1.0.0"
+
+type TestProto = ref object of LPProtocol
+```
+
+We've set a [protocol ID](https://docs.libp2p.io/concepts/protocols/#protocol-ids), and created a custom `LPProtocol`. In a more complex protocol, we could use this structure to store interesting variables.
+
+A protocol generally has two part: and handling/server part, and a dialing/client part.  
+Theses two parts can be identical, but in our trivial protocol, the server will wait for a message from the client, and the client will send a message, so we have to handle the two cases separately.
+
+Let's start with the server part:
+```nim
+proc new(T: typedesc[TestProto]): T =
+  # every incoming connections will in handled in this closure
```

Comment:
Looks like a typo? `every incoming connections will **be** handled in this closure`?